File: mario-less/mario.c
// Prints right-aligned pyramid
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{
    // Promt the user for pyramid height
    int height;
    do
    {
        height = get_int("Please enter the height: ");
    }
    while (height < 1);

    for (int r = 1; r <= height; r++)
    {
        print_row(height - r, r);
    }
}

void print_row(int spaces, int bricks)
{
    for (int j = 1; j <= spaces; j++)
    {
        printf(" ");
    }
    for (int i = 1; i <= bricks; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: sentimental-mario-less/mario.py
# Program to print half-pyramid

from cs50 import get_int


def main():

    # Promts user for height
    while True:
        height = get_int("Enter Height: ")
        if height >= 0 and height <= 8:
            break

    # Prints the pyramid with given height
    for r in range(height):
        print_row(height - (r + 1), r + 1)


# Print row of bricks(right aligned)
def print_row(spaces, bricks):

    # Print spaces
    for s in range(spaces):
        print(" ")

    # Print bricks
    for b in range(bricks):
        print("#")

    print("")

main()

